File: CollaborationBoard/Source/Actions/ActionGroup.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.Web;

namespace CollaborationBoard
{
    [DataContract]
    public class ActionGroup
    {
        [DataMember(Name = "user")]
        public string User { get; set; }

        [DataMember(Name = "messages")]
        public List<Message> Messages { get; set; }

        [DataMember(Name = "lines")]
        public List<Line> Lines { get; set; }

        public bool Empty
        {
            get
            {
                return this.Messages.Count == 0 && this.Lines.Count == 0;
            }
        }

        public ActionGroup()
        {
            this.Messages = new List<Message>();
            this.Lines = new List<Line>();
        }
    }
}

File: CollaborationBoard/Scripts/App/actionGroup.js
(function () {
    "use strict";

    window.Collab = window.Collab || new Object();

    var actionGroup = function (messages, lines) {
        this.Messages = messages;
        this.Lines = lines;
    };

    actionGroup.empty = new actionGroup([], []);

    var point = function (x, y) {
        this.X = x || 0;
        this.Y = y || 0;
    };

    var line = function (points) {
        this.Points = points || [];
    };

    var message = function (text) {
        this.text = text || "";
    };

    Collab.Point = point;
    Collab.Line = line;
    Collab.Message = message;
    Collab.ActionGroup = actionGroup
})();